stream-chat-android/src/main/java/com/getstream/sdk/chat/utils/CollectionsUtils.kt

```diff
@@ -0,0 +1,34 @@
+package com.getstream.sdk.chat.utils
+
+// Use when it is known that the element is always close to the end
+internal fun <T> List<T>.reverseFind(predicate: (T) -> Boolean): T? {
+    for (i in (this.size - 1) downTo 0) {
+        if (predicate(this[i])) {
+            return this[i]
+        }
+    }
+
+    return null
+}
+
+// Use when it is known that the element is always close to the end
+internal fun <T> List<T>.reverseIndexOf(element: T): Int {
```

Comment:
Same here:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/last-index-of.html